examples/seismic/abc_methods/01_introduction.ipynb

```diff
@@ -0,0 +1,1192 @@
+   "source": [
+    "src_term = src.inject(field=u.forward,expr=src*dt**2*vel0**2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To define the Receivers we use the structure called *rec.interpolate*. We need to define the expression *expr* to\n",
+    "be computed at the receivers location, which in this case is the value of the displacement.\n",
+    "\n",
+    "In this way, the Receivers term is named *rec_term* and is created as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rec_term = rec.interpolate(expr=u)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now create the acoustic equation without the external force term. As defined previusly, *u* represents the displacement field and *vel0* the field that carries the velocity information. \n",
+    "\n",
+    "- The structure that creates Equations in Devito is *Eq(eq)*, where *eq* is the equation we want to assign;\n",
+    "- .dt2 calculates the second temporal derivative of the field it is applied to;\n",
+    "- .laplace builds the spatial Laplacian operator of a given field. In the present case, this operator is .dx2 + .dz2;\n",
+    "\n",
+    "The acoustic equation without the source term has the following form:\n",
+    "\n",
+    "- $ u_ {tt} - c ^ {2} (u_ {xx} + u_ {zz}) $\n",
+    "\n",
+    "Using Devito's notation, we have\n",
+    "\n",
+    "- eq = u.dt2 - vel0 * vel0 * u.laplace\n",
+    "\n",
+    "Then, using the operator *Eq(eq)*  we create the *pde* that represents the acoustic equation without the source term:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pde = Eq(u.dt2 - u.laplace*vel0*vel0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For Devito to solve a predefined *pde* we use the expression *solve(pde,field)* where the parameters are the equation we want to solve (*pde*) and the field being  updated (*field1*). In our case, we want to solve the *pde* in the *u.forward* field, so we have\n",
+    "\n",
+    "- solve(pde,u.forward)\n",
+    "\n",
+    "Now we need Devito to assign this solution to a field, where we use the command *Eq()* again, with a few more options: more precisely we will have *Eq(field2,eq2,subdomain)*, where *field2* is the field who will receive the solution of *eq2* computed in the given *subdomain*. Since the only subdomain we have is *d0*,we define the expression of the subdomain as:\n",
+    "\n",
+    "- subdomain=grid.subdomains ['d0']\n",
+    "\n",
+    "Altogether we define the *stencil* for the acoustic problem:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "stencil  =  Eq(u.forward, solve(pde,u.forward),subdomain = grid.subdomains['d0'])"
```

Comment:
Maybe worth showing the mathjax representation of your stencil to show it corresponds with your maths above.

Reply:
I didn't understand your suggestion in this comment. Could you include some more details?

Reply:
@rhodrin Here is it ok?!